services/auth/auth-supabase.service.ts

```diff
@@ -11,37 +12,44 @@ class AuthService {
     this.supabase = createClient(supabaseUrl, supabaseAnonKey);
   }
 
-  private handleError(error: any): { user: null; token: undefined } {
-    errorHandler(error.message);
-    return { user: null, token: undefined };
-  }
-
   async registerUser(
     email: string,
     password: string,
-  ): Promise<{ user: User | null; token: string | undefined }> {
+  ): Promise<{ user: User | null; token: string | undefined } | null> {
     const { data, error } = await this.supabase.auth.signUp({
       email,
       password,
     });
-    if (error) return this.handleError(error);
+    if (error) return errorHandler(error.message);
 
     return { user: data.user, token: data.session?.access_token };
   }
 
   async loginUser(
     email: string,
     password: string,
-  ): Promise<{ user: User | null; token: string | undefined }> {
+  ): Promise<{ user: User | null; token: string | undefined } | null> {
     const { data, error } = await this.supabase.auth.signInWithPassword({
       email,
       password,
     });
-    if (error) return this.handleError(error);
+    if (error) return errorHandler(error.message);
 
     return { user: data.user, token: data.session?.access_token };
   }
 
+  async oauth(provider: Provider): Promise<{provider: Provider; url: string;} | null> {
+    const { data, error } = await this.supabase.auth.signInWithOAuth({
+      provider: provider,
+      options: {
+        redirectTo: `${window.location.origin}/app/dashboard`
```

Comment:
here `dashboard` is a registered component. Lets have the consts in  CORE  and here `/app/${MODULE.DASHBOARD"`